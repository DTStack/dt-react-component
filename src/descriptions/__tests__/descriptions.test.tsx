import React from 'react';
import { render } from '@testing-library/react';

import Descriptions from '..';

const TestDescriptions = ({ fixed }: { fixed?: boolean }) => {
    return (
        <Descriptions bordered column={2} tableLayout={fixed ? 'fixed' : 'auto'}>
            <Descriptions.Item label="名称">我是 dt-react-component 组件库</Descriptions.Item>
            <Descriptions.Item label="描述">
                <div
                    style={{
                        width: 0,
                        textOverflow: 'ellipsis',
                        overflow: 'hidden',
                        whiteSpace: 'nowrap',
                        minWidth: '100%',
                    }}
                >
                    很长很长的描述描述描述描述描述描述描述描述描述描述描述描述描述描述描述描述描述，长度甚至超过了一行
                </div>
            </Descriptions.Item>
        </Descriptions>
    );
};

describe('Test Descriptions', () => {
    it('match snapshot', () => {
        expect(render(<TestDescriptions />).asFragment()).toMatchSnapshot(
            "As same as antd's Descriptions"
        );

        expect(render(<TestDescriptions fixed />).asFragment()).toMatchSnapshot(
            'Fixed table layout'
        );
    });
});
