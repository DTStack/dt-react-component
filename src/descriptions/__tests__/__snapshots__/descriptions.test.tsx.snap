// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Descriptions match snapshot: As same as antd's Descriptions 1`] = `
<DocumentFragment>
  <div
    class="ant-descriptions ant-descriptions-bordered"
  >
    <div
      class="ant-descriptions-view"
    >
      <table>
        <tbody>
          <tr
            class="ant-descriptions-row"
          >
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span>
                名称
              </span>
            </th>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                我是 dt-react-component 组件库
              </span>
            </td>
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span>
                描述
              </span>
            </th>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                <div
                  style="width: 0px; text-overflow: ellipsis; overflow: hidden; white-space: nowrap; min-width: 100%;"
                >
                  很长很长的描述描述描述描述描述描述描述描述描述描述描述描述描述描述描述描述描述，长度甚至超过了一行
                </div>
              </span>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Test Descriptions match snapshot: Fixed table layout 1`] = `
<DocumentFragment>
  <div
    class="ant-descriptions ant-descriptions-bordered dtc-descriptions__fixed"
  >
    <div
      class="ant-descriptions-view"
    >
      <table>
        <tbody>
          <tr
            class="ant-descriptions-row"
          >
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span>
                名称
              </span>
            </th>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                我是 dt-react-component 组件库
              </span>
            </td>
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span>
                描述
              </span>
            </th>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                <div
                  style="width: 0px; text-overflow: ellipsis; overflow: hidden; white-space: nowrap; min-width: 100%;"
                >
                  很长很长的描述描述描述描述描述描述描述描述描述描述描述描述描述描述描述描述描述，长度甚至超过了一行
                </div>
              </span>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;
